src/AutoRest.CSharp/Mgmt/Generation/ResourceGroupExtensionsWriter.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using AutoRest.CSharp.AutoRest.Plugins;
+using AutoRest.CSharp.Input;
+using AutoRest.CSharp.Mgmt.Decorator;
+using AutoRest.CSharp.Mgmt.Output;
+using AutoRest.CSharp.Output.Models;
+using AutoRest.CSharp.Output.Models.Types;
+using Azure.Core.Pipeline;
+using Azure.ResourceManager.Core;
+
+namespace AutoRest.CSharp.Generation.Writers
+{
+    internal class ResourceGroupExtensionsWriter
+    {
+        protected string Description = "A class to add extension methods to ResourceGroup.";
+        protected string Accessibility = "public";
+        protected string Type = "ResourceGroupExtensions";
+
+        public void WriteExtension(string @namespace, MgmtConfiguration mgmtConfiguration, CodeWriter writer, Dictionary<OperationGroup, Resource> armResources)
+        {
+            using (writer.Namespace(@namespace))
+            {
+                writer.WriteXmlDocumentationSummary(Description);
+                using (writer.Scope($"{Accessibility} static class {Type}"))
```

Comment:
partial

all types we generate should be partial

Reply:
Done